File: app/src/main/java/de/bikebean/app/db/settings/settings/WappState.java
package de.bikebean.app.db.settings.settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.sms.SmsFactory;
import de.bikebean.app.db.state.State;
import de.bikebean.app.ui.initialization.StateList;
import de.bikebean.app.ui.drawer.status.location.LocationStateViewModel;

public class WappState extends Wapp {

    private final @NonNull State wifiAccessPoints;
    private final @NonNull State cellTowers;

    public WappState(final LocationStateViewModel st, final @NonNull StateList states) {
        super(SmsFactory.createSmsFromState(states.getCellTowersState(st)), 0.0);

        wifiAccessPoints = states.getWifiAccessPointsState(st);
        cellTowers = states.getCellTowersState(st);
    }

    public int getSmsId() {
        return cellTowers.getSmsId();
    }

    public boolean getIsNull() {
        return cellTowers.getIsNull() || wifiAccessPoints.getIsNull();
    }

    public @NonNull State getCellTowers() {
        return cellTowers;
    }

    public @NonNull State getWifiAccessPoints() {
        return wifiAccessPoints;
    }

    public boolean getIfNewest(final @NonNull LocationStateViewModel st) {
        return (wifiAccessPoints.equalsId(st.getConfirmedLocationSync(wifiAccessPoints))
                && cellTowers.equalsId(st.getConfirmedLocationSync(cellTowers)));
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/WappState.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings

import de.bikebean.app.db.sms.SmsFactory
import de.bikebean.app.db.state.State
import de.bikebean.app.db.state.StateFactory
import de.bikebean.app.ui.drawer.status.location.LocationStateViewModel

class WappState : Wapp {

    val wifiAccessPoints: State
    val cellTowers: State

    constructor(_cellTowers: State, _wifiAccessPoints: State) :
            super(SmsFactory.createSmsFromState(_cellTowers), 0.0) {
        wifiAccessPoints = _wifiAccessPoints
        cellTowers = _cellTowers
    }

    constructor(_cellTowers: State) :
            super(SmsFactory.createSmsFromState(_cellTowers), 0.0) {
        wifiAccessPoints = StateFactory.createNullState()
        cellTowers = StateFactory.createNullState()
    }

    val smsId: Int
        get() = cellTowers.smsId

    val isNull: Boolean
        get() = cellTowers.isNull || wifiAccessPoints.isNull

    fun getIfNewest(st: LocationStateViewModel): Boolean {
        return (wifiAccessPoints.equalsId(st.getConfirmedLocationSync(wifiAccessPoints))
                && cellTowers.equalsId(st.getConfirmedLocationSync(cellTowers)))
    }

}